server/chat-plugins/poll.js

```diff
@@ -273,50 +273,50 @@ const commands = {
 		newhelp: [`/poll create [question], [option1], [option2], [...] - Creates a poll. Requires: % @ # & ~`],
 
 		vote(target, room, user) {
-			if (!room.poll) return this.errorReply("There is no poll running in this room.");
+			if (!room.pollOrAnnouncement || !(room.pollOrAnnouncement instanceof Poll)) return this.errorReply("There is no poll running in this room.");
 			if (!target) return this.parse('/help poll vote');
 
 			if (target === 'blank') {
-				room.poll.blankvote(user);
+				room.pollOrAnnouncement.blankvote(user);
 				return;
 			}
 
 			let parsed = parseInt(target);
 			if (isNaN(parsed)) return this.errorReply("To vote, specify the number of the option.");
 
-			if (!room.poll.options.has(parsed)) return this.sendReply("Option not in poll.");
+			if (!room.pollOrAnnouncement.options.has(parsed)) return this.sendReply("Option not in poll.");
 
-			room.poll.vote(user, parsed);
+			room.pollOrAnnouncement.vote(user, parsed);
 		},
 		votehelp: [`/poll vote [number] - Votes for option [number].`],
 
 		timer(target, room, user) {
-			if (!room.poll) return this.errorReply("There is no poll running in this room.");
+			if (!room.pollOrAnnouncement || !(room.pollOrAnnouncement instanceof Poll)) return this.errorReply("There is no poll running in this room.");
```

Comment:
You can use whatever check you want, as long as you do 

`const poll = /** @type {Poll} */ (room.pollOrAnnouncement);`